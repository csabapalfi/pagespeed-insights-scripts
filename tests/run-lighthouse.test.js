const {launch} = require('chrome-launcher');
const lighthouse = require('lighthouse');
const runLighthouse = require('../lib/run-lighthouse');

jest.mock('chrome-launcher');
jest.mock('lighthouse');

describe('run-lighthouse', () => {
  it('runLightHouse launches chrome and runs Lighthouse', async () => {
    const url = 'https://www.google.com';
    const lighthouseResult = {};
    const lighthouseArtifacts = {};
    const options = {cpuSlowDown: 4};
    const chrome = {port: 1234, kill: jest.fn()};
    launch.mockResolvedValue(chrome);
    lighthouse.mockResolvedValue({
      lhr: lighthouseResult, 
      artifacts: lighthouseArtifacts
    });

    const {result, artifacts} = await runLighthouse(options, url);

    expect(result).toBe(lighthouseResult);
    expect(artifacts).toBe(lighthouseArtifacts);

    expect(launch).toHaveBeenCalledTimes(1);
    expect(launch.mock.calls[0]).toMatchSnapshot();

    expect(lighthouse).toHaveBeenCalledTimes(1);
    expect(lighthouse.mock.calls[0]).toMatchSnapshot();

    expect(chrome.kill).toHaveBeenCalledTimes(1);
  });
});