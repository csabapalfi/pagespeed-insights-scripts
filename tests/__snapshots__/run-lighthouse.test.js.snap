// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`run-lighthouse runLightHouse launches chrome and runs Lighthouse 1`] = `
Array [
  Object {
    "chromeFlags": Array [
      "--headless",
    ],
  },
]
`;

exports[`run-lighthouse runLightHouse launches chrome and runs Lighthouse 2`] = `
Array [
  "https://www.google.com",
  Object {
    "port": 1234,
    "throttling": Object {
      "cpuSlowdownMultiplier": 4,
    },
  },
  Object {
    "extends": "lighthouse/lighthouse-core/config/lr-mobile-config.js",
    "settings": Object {
      "onlyAudits": Array [
        "first-contentful-paint",
        "first-meaningful-paint",
        "speed-index",
        "first-cpu-idle",
        "interactive",
        "time-to-first-byte",
        "user-timings",
      ],
    },
  },
]
`;
